### PR TITLE
framework/st_things   :  Change the check management for readonly exist

### PR DESCRIPTION
**Pull request by uvasistha-samsung**
framework/src/st_things : Change the check management for readonly exist

Change the "if statement" logic to check whether a read only attribute is present or not in the request.
signed off by: uvasistha-samsung  utkarsh.m@samsung.com

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [12b72af73709081f54c564ee8104fe5bec926d9a] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [12b72af73709081f54c564ee8104fe5bec926d9a] - Code Rule Check (C++) OK.

**Comment by uvasistha-samsung** [comment]
@sunghan-chang  @Hyunjun85
 Are the changes to tittle and description fine ?

**Comment by seinfra** [comment]
Target : [60d8c7e4d3474853a85b94585a2a2b7fec98a199] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [60d8c7e4d3474853a85b94585a2a2b7fec98a199] - Code Rule Check (C++) OK.

File: framework/src/st_things/st_things_request_handler.c [hunk 1]
```diff
@@ -977,7 +977,7 @@ bool handle_post_req_helper(const char *res_uri, const char *query, OCRepPayload
 		}
 
 		if (!IS_WRITABLE(properties[index]->rw)) {
-			if (OCRepPayloadIsNull(req_payload, properties[index]->key)) {
+			if (!OCRepPayloadIsNull(req_payload, properties[index]->key)) {
 				THINGS_LOG_E(TAG, "(%s) is not present in the request.", properties[index]->key);
```
Comment (JeongJunSik) [inline framework/src/st_things/st_things_request_handler.c]: please change the log message also.

**Comment by seinfra** [comment]
Target : [1cee6bb6e670fedfc27a793b3feb5788ec135a4f] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [1cee6bb6e670fedfc27a793b3feb5788ec135a4f] - Code Rule Check (C++) OK.

**Comment by uvasistha-samsung** [comment]
@JeongJunSik 
Updated the PR : changed the log 
Please review

**Comment by uvasistha-samsung** [comment]
@Hyunjun85 Please review and approve 
@JeongJunSik Dismiss the stale review

**Pull request merged**